ocp_resources/datavolume.py

```diff
@@ -223,13 +219,59 @@ def _check_none_pending_status(self, failure_timeout=120):
             for sample in TimeoutSampler(
                 wait_timeout=failure_timeout,
                 sleep=10,
-                func=lambda: self.instance.status.phase,
+                func=lambda: self.exists,
             ):
                 # If DV status is Pending (or Status is not yet updated) continue to wait, else exit the wait loop
-                if sample in [self.Status.PENDING, None]:
+                if sample and (
+                    not sample.status
+                    or sample.status.phase
+                    in [
+                        self.Status.PENDING,
+                        None,
+                    ]
+                ):
                     continue
-                else:
-                    break
+                break
+        except TimeoutExpiredError:
+            self.logger.error(f"{self.name} status is {sample}")
+            raise
+
+    def wait_for_dv_success(self, timeout=600, failure_timeout=120):
+        self.logger.info(f"Wait DV success for {timeout} seconds")
+        self._check_none_pending_status(failure_timeout=failure_timeout)
+        self.pvc.wait_for_status(
+            status=PersistentVolumeClaim.Status.BOUND, timeout=timeout
+        )
+        # if garbage collector enabled, the DV deleted once succeeded
+        try:
+            for sample in TimeoutSampler(
+                sleep=1,
+                wait_timeout=timeout,
+                func=lambda: self.exists,
+            ):
+                # DV reach to success if the status is succeeded or if the DV does not exists
```

Comment:
Please add to your comment why we accept that the DV is not exists (garbage collector)

Reply:
done